packages/monitor-v2/src/monitor-og/oSnapAutomation.ts

```diff
@@ -723,7 +741,12 @@ export const proposeTransactions = async (logger: typeof Logger, params: Monitor
   const supportedProposals = await getSupportedSnapshotProposals(logger, supportedModules, params);
 
   // Get all undiscarded on-chain proposals for supported modules.
-  const onChainProposals = await getUndiscardedProposals(Object.keys(supportedModules), params);
+  let onChainProposals;
```

Comment:
this is used not only in `potentialProposals`, but also in `unblockedProposals`. I think each requires different approach, otherwise this could skip proposals for another snapshot vote with matching tx content that had been disputed.

Reply:
I believe #4768 should address this